SPECS-CROSS/glibc-bootstrap_cross/glibc-bootstrap2_cross.spec

```
@@ -218,21 +208,30 @@ cd %{_builddir}/%{name}-build
             --prefix=/ \
             --build=%{_host_arch}-%{_vendor}-linux-gnu \
             --host=%{_tuple} \
-            --target=%{_tuple} \
             --with-sysroot="$TEMP_SYSROOT" \
             --with-headers="$TEMP_SYSROOT/%{_includedir}" \
             --disable-multilib \
             libc_cv_forced_unwind=yes \
             libc_cv_ctors_header=yes \
-            --disable-werror
+            --disable-werror \
+            --enable-languages=c,c++ \
+            libc_cv_c_cleanup=yes \
+            CXX=arch64-mariner-linux-gnu-gcc
```

Without this arg, the build fails with `"/opt/cross/aarch64-mariner-linux-gnu/lib/gcc/aarch64-mariner-linux-gnu/11.2.0/../../../../aarch64-mariner-linux-gnu/bin/ld: /usr/src/mariner/BUILD/aarch64-mariner-linux-gnu-glibc-bootstrap2-build/support/links-dso-program.o: error adding symbols: file in wrong format"`

---

```suggestion
            CXX=aarch64-mariner-linux-gnu-gcc
```
Interesting finding! Just curious, how does the `CXX=` option work here? Any docs that help describe its usage?

---

Great catch! Yes this is a typo. According to the source code, the CXX represents the environment variable for the "C++ compiler command". This can be found in the 'configure' script after downloading https://ftp.gnu.org/gnu/glibc/glibc-2.35.tar.xz